src/components/Auth/Login.tsx

```diff
@@ -7,27 +7,67 @@ import {
   messageType,
 } from 'lib/flashMessage';
 import PropTypes from 'prop-types';
-import React from 'react';
-import { connect } from 'react-redux';
+import React, { ReactNode } from 'react';
+import { connect, DispatchProp } from 'react-redux';
 import { Link } from 'react-router-dom';
-import { bindActionCreators } from 'redux';
+import { bindActionCreators, Dispatch } from 'redux';
 import { AppRoutes } from 'shared/constants/routes';
 import SlideTransition from 'styles/transitions/SlideTransition';
 import Button from 'UI/Button';
 
 import { parseErrorMessages } from './parseErrorMessages';
 
-class Login extends React.Component {
-  state = {
+// The props coming from the global state (AKA: `mapStateToProps`)
+interface IStateProps {
+  // For now, until we have a type for it
+  user: Record<string, never>;
+  // For now, until we have a type for it
+  flashMessages: Record<string, never>;
+}
+
+// The props coming from injected actions (AKA: `mapDispatchToProps`)
+interface IDispatchProps extends DispatchProp {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  actions: Record<string, (...args: any[]) => Promise<any>>;
+}
+
+interface ILoginProps extends IStateProps, IDispatchProps {}
+
+interface ILoginState {
+  email: string;
+  password: string;
+  authenticating: boolean;
+}
+
+class Login extends React.Component<ILoginProps, ILoginState> {
+  public static propTypes: IStateProps & IDispatchProps = {
```

Comment:
`public` is not the default label? Should we label everything as `public` then?

Reply:
`public` is indeed the default label. I prefer to specifically write it for consistency, since we're writing `private` and `protected`, as well